addon/@ember/test-waiters/wait-for.ts

```diff
@@ -1,18 +1,25 @@
 import { DEBUG } from '@glimmer/env';
 import waitForPromise from './wait-for-promise';
+import buildWaiter from './build-waiter';
 import { PromiseType } from './types';
 
+type CoroutineGenerator<T> = Generator<any, T, any>;
 type AsyncFunction<A extends Array<any>, PromiseReturn> = (...args: A) => Promise<PromiseReturn>;
-type AsyncFunctionArguments = [AsyncFunction<any[], any>, string?];
+type TaskFunction<A extends any[], T> = (...args: A) => CoroutineGenerator<T>;
```

Comment:
No, we're not specifically thinking of/limiting to `ember-concurrency` here. I was thinking of `task` in the completely generic sense of a thing you're trying to do. That's also the language `ember-concurrency` uses, but it obviously didn't invent that concept, so I wasn't thinking of using `task` as explicitly referring to `ember-concurrency` -- just to a concept that it also uses. That being said, I chose it with fairly minimal thought, so am definitely open to different naming. `GeneratorFunction` is already a type (an interface), so I don't think that's a good one to use -- I suppose `CoroutineGeneratorFunction` would make sense since it's consistent with `CoroutineGenerator`.

Reply:
Ya I personally feel like if we're sticking to a taxonomy for our internal branches, we should keep them consistent. I'm 👍 for the names you suggested.